File: 3월/20200330/python/who-has-not-finished_test.py
import collections


def solution(participant, completion):
    answer = collections.Counter(participant) - collections.Counter(completion)
    return list(answer.keys())[1]


def test_solution():
    assert solution(['leo', 'kiki', 'eden'], ['eden', 'kiki']) == 'leo'
    assert solution(['marina', 'josipa', 'nikola', 'vinko', 'filipa'],
                    ['josipa', 'filipa', 'marina', 'nikola']) == 'vinko'
    assert solution(['mislav', 'stanko', 'mislav', 'ana'],
                    ['stanko', 'ana', 'mislav']) == 'mislav'

File: 03-March/20200330/javascript/완주하지_못한_선수/whoHasNotFinished.test.js
const solution = (participant, completion) => {
  const participants = [...participant].sort();
  const completions = [...completion].sort();
  return participants.find((it, index) => it !== completions[index]);
};

test("solution", () => {
  expect(solution(["leo", "kiki", "eden"], ["eden", "kiki"])).toEqual("leo");
  expect(
    solution(
      ["marina", "josipa", "nikola", "vinko", "filipa"],
      ["josipa", "filipa", "marina", "nikola"]
    )
  ).toEqual("vinko");
  expect(
    solution(["mislav", "stanko", "mislav", "ana"], ["stanko", "ana", "mislav"])
  ).toEqual("mislav");
});
